spec/controllers/admin/bulk_affiliate_deactivate_controller_spec.rb

```diff
@@ -0,0 +1,115 @@
+require 'spec_helper'
+
+describe Admin::BulkAffiliateDeactivateController, type: :controller do
+  fixtures :users
+  let(:user) { users('affiliate_admin') }
+
+  before { activate_authlogic }
+
+  describe "GET #index" do
+    context 'when not logged in' do
+      it 'redirects to the login page' do
+        get :index
+        expect(response).to redirect_to login_path
+      end
+    end
+
+    context 'when logged in as an admin' do
+      before do
+        UserSession.create(user)
+      end
+
+      it 'renders the index template successfully' do
+        get :index
+        expect(response).to be_successful
+        expect(response).to render_template(:index)
+      end
+
+      it 'assigns the page title' do
+        get :index
+        expect(assigns(:page_title)).to eq('Bulk Affiliate Deactivate')
+      end
+    end
+  end
+
+  describe 'POST #upload' do
+    let(:file) { fixture_file_upload('/csv/deactivate_affiliates.csv', 'text/csv') }
+    let(:upload_with_file) { post :upload, params: { file: file } }
```

Comment:
`Let`'s are for variables not for execution any actions. What do you think about adding a context, e.g. "When admin uploads a file and do the post in a `before` block.